ibis/common/selectors.py

```diff
@@ -29,3 +28,79 @@ def expand(self, table: ir.Table) -> Sequence[ir.Value]:
             A sequence of value expressions that match the selector
 
         """
+
+    @abc.abstractmethod
+    def expand_names(self, table: ir.Table) -> frozenset[str]:
+        """Compute the set of column names that match the selector."""
+
+
+class Selector(Expandable):
+    """A column selector."""
+
+    def expand(self, table: ir.Table) -> Sequence[ir.Value]:
+        names = self.expand_names(table)
+        return list(map(table.__getitem__, filter(names.__contains__, table.columns)))
```

Comment:
General comment - IMO `map`/`filter` are (usually) an antipattern, the list comprehension version of this is both shorter and reads a lot cleaner to me.

```suggestion
        return [table[n] for n in table.columns if n in names]
```

Reply:
I don't consider `map` and `filter` an anti-pattern, I've never really understood any of the arguments for avoiding them.

Reply:
Here, avoiding the overhead of iterating in Python (I realize this might not be huge in most cases) seemed like an acceptable trade-off given I personally didn't consider one to be horribly worse.

Happy to change it though.

Reply:
In this case at least using a comprehension removes the need for writing out two magic method names, which is an improvement. IMO `map`/`filter`/`reduce` do have their place, but in many cases I still prefer the comprehension. Up to you, not going to argue more here.

Reply:
Fair enough, the magic method names are a little annoying 😅